### PR TITLE
[Lint Warnings] Source Sets: Debug, WordPress and Jetpack - Resolve All Warnings

### LINKED ISSUE #18192: Lint Warnings - Source Sets: Debug, WordPress and Jetpack [`All`]
Parent #18162

This issue is about resolving/suppressing the Lint warnings on all other source sets that are not part of the `main` source set.

Number of Warnings: `15`

- [x] Debug `2`
    - [x] Correctness `2`
        - [x] ScopedStorage `1`
        - [x] ManifestOrder `1`
- [x] WordPress `3`
    - [x] Security `1`
        - [x] ExportedContentProvider `1`
    - [x] Performance `2`
        - [x] VectorPath `2`
- [x] Jetpack `10`
    - [x] Correctness `9`
        - [x] LocaleFolder `2`
        - [x] IntentFilterUniqueDataAttributes `6`
        - [x] ModifierParameter `1`
    - [x] Performance `1`
        - [x] VectorPath `1`

### PR DESCRIPTION
**Pull request by ParaskP7**
Parent #18162
Closes #18192

This PR resolves all Lint related warnings for the `Debug`, `WordPress` and `Jetpack` source sets:

-----

Reconfigure Lint (Ignore):

1. [Ignore locale folder lint warnings for jetpack.](https://github.com/wordpress-mobile/WordPress-Android/commit/c272690c89b81589184cf1b56e0bfe5a0d43e90a)

Warnings Resolution List:

1. [Resolve manifest order lint warning for debug.](https://github.com/wordpress-mobile/WordPress-Android/commit/80e895d1a2926b0840c3d2ff0c2a8ca98ae257ce)
2. [Resolve intent filter unique data attributes jetpack warnings.](https://github.com/wordpress-mobile/WordPress-Android/commit/11172969a44dc48f495b97f300d4062520bbf6a2)
3. [Resolve modifier parameter lint warnings for jetpack.](https://github.com/wordpress-mobile/WordPress-Android/commit/bbbddb1b3cad67a2f9c9453b39831075588765c2)

Warnings Suppression List:

1. [Suppress scoped storage lint warning for debug.](https://github.com/wordpress-mobile/WordPress-Android/commit/0ce9dc2a0f86c1ad977c6b0c5948df67237d37be)
2. [Suppress missing application icon lint warning for debug.](https://github.com/wordpress-mobile/WordPress-Android/commit/456eb64bfc7e27d289e919bdda353786397a8991)
3. [Suppress exported content provider lint warning for wordpress.](https://github.com/wordpress-mobile/WordPress-Android/commit/57c07666583162531db6a120f7107dae9c4ce753)
4. [Suppress vector path lint warnings for wordpress/jetpack.](https://github.com/wordpress-mobile/WordPress-Android/commit/1c51abfe6d12f1fbf01fe084b323958643aa41cf)
5. [Suppress missing application icon lint warning for jetpack.](https://github.com/wordpress-mobile/WordPress-Android/commit/015947c175bbd3a75b3bcd202caf828e581fca30)

Documentation List:

1. [Add missing todo for missing firebase instance token refresh.](https://github.com/wordpress-mobile/WordPress-Android/commit/3b659fb01fc2046520436d2543c9779631150672)
2. [Remove unnecessary main section from lint baseline.](https://github.com/wordpress-mobile/WordPress-Android/commit/7d3dda173ff7482512c206ba450a016602c941c3)

-----

PS: @ovitrif I added you as the main reviewer, randomly, since I just wanted someone from the WordPress team to be aware of and sign-off on that change for WPAndroid. I also added the @wordpress-mobile/apps-infrastructure team, but this in done only for monitoring purposes, as such, I am not expecting any active review from that team. Thus, feel free to merge this PR if you deem so.

-----

To test:

- There is nothing much to test here.
- Verifying that all the CI checks are successful should be enough.
- However, if you want to be thorough about reviewing this change, you could quickly smoke test both, the WordPress and Jetpack apps, and see if they are both working as expected.

FYI: As far as I can tell, the only commit that might cause some behavior change might be this 11172969a44dc48f495b97f300d4062520bbf6a2 commit. Thus, you might want to check this one out in a bit more detail/depth. Everything else shouldn't cause any trouble. 

-----

## Regression Notes
1. Potential unintended areas of impact

    - Potential breakage or misbehaviour of the Jetpack related `<intent-filter>` functionality on `DeepLinkingIntentReceiverActivity` and its `android.intent.action.VIEW` action (see 11172969a44dc48f495b97f300d4062520bbf6a2 commit).

6. What I did to test those areas of impact (or what existing automated tests I relied on)

    - See `To test` section above.

7. What automated tests I added (or what prevented me from doing so)

    - N/A

-----

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/jetpack/java/org/wordpress/android/ui/accounts/login/components/RepeatingColumn.kt [hunk 1]
```diff
@@ -17,9 +17,9 @@ import androidx.compose.ui.unit.Constraints
  */
 @Composable
 fun RepeatingColumn(
+    modifier: Modifier = Modifier,
```
Comment (ovitrif) [inline WordPress/src/jetpack/java/org/wordpress/android/ui/accounts/login/components/RepeatingColumn.kt]: AFAIK the rule for `modifier` is to be the first optional param (I.e. having a default value)( 👍🏻 
Would you mind moving it below `position` please? 🙇‍♂️

**Review by ovitrif — APPROVED** [review]
Thank you for this awesome cleanup work @ParaskP7 🙇🏻 

I tested the functionality that might have been affected by commit https://github.com/wordpress-mobile/WordPress-Android/commit/11172969a44dc48f495b97f300d4062520bbf6a2 using my public codepen: [WP Deep Links Helper](https://codepen.io/ovitrif/full/WNKRBeV). All links worked as expected in the JP app 🎉 

I left a small np comment on composables params ordering, but feel free to ignore it if you don't 💯 agree, and :shipit: 🚀

File: WordPress/src/jetpack/java/org/wordpress/android/ui/accounts/login/components/RepeatingColumn.kt (hunk 1 above)
Comment (ParaskP7) [reply WordPress/src/jetpack/java/org/wordpress/android/ui/accounts/login/components/RepeatingColumn.kt]: Great, yes, this makes sense @ovitrif , totally agree, thanks for noticing and [done](https://github.com/wordpress-mobile/WordPress-Android/pull/18227/commits/d7434f571b2737b082f82563ac76a44f82abc3bc)! 🙇 💯 ✅

**Comment by ParaskP7** [comment]
Thank you so much for the review and testing @ovitrif , you are awesome! 🙇 ❤️ 🚀

> I tested the functionality that might have been affected by commit https://github.com/wordpress-mobile/WordPress-Android/commit/11172969a44dc48f495b97f300d4062520bbf6a2 using my public codepen: [WP Deep Links Helper](https://codepen.io/ovitrif/full/WNKRBeV). All links worked as expected in the JP app 🎉

🎉 Thank YOU! 🙇 x 💯

File: WordPress/src/jetpack/java/org/wordpress/android/ui/accounts/login/components/RepeatingColumn.kt (hunk 1 above)
Comment (ovitrif) [reply WordPress/src/jetpack/java/org/wordpress/android/ui/accounts/login/components/RepeatingColumn.kt]: Awesome, thanks a bunch for applying it 🥇 🚀

**Pull request merged**